### ISSUE TITLE
`SetGlobal` is not working after lua vm is recycled

### ISSUE THREAD

**Issue by NKUCodingCat** [ZenLiuCN/glu#1]
I am trying to use glu to recycle lua vm and using `GetGlobal` / `SetGlobal` to access lua global variants in Go. Howerver, some counterintuitive behavior were found.

Considering this unit test:
```
func TestGlu(t *testing.T) {
	vmPool := glu.CreatePool()
	vm := vmPool.Get()
	assert.NotNil(t, vm)
	vmaddr := fmt.Sprintf("%p", vm)

	assert.Equal(t, lua.LNil, vm.GetGlobal("__SOME_KEY__"))       // Passed
	assert.NoError(t, vm.DoString("assert(__SOME_KEY__ == nil)")) // Passed, __SOME_KEY__ is nil in and out of lua
	vm.SetGlobal("__SOME_KEY__", lua.LString("ZZZZZZ"))
	assert.Equal(t, lua.LString("ZZZZZZ"), vm.GetGlobal("__SOME_KEY__")) // Passed
	assert.NoError(t, vm.DoString("assert(__SOME_KEY__ == 'ZZZZZZ')"))   // Passed, __SOME_KEY__ is same in and out of lua

	vmPool.Put(vm) // Recycle the vm
	vm = vmPool.Get()
	vmaddr2 := fmt.Sprintf("%p", vm)
	assert.Equal(t, vmaddr, vmaddr2) // vm is actually recycled

	// Something strange happened....
	// Global is not reset after recycle
	assert.Equal(t, lua.LNil, vm.GetGlobal("__SOME_KEY__")) // Assertion failed, Value still exists out of lua
	// But I cannot find it in lua, __SOME_KEY__ is nil
	assert.NoError(t, vm.DoString("assert(__SOME_KEY__ == nil)")) // Passed, Nothing in lua

	// What if I set it again?
	vm.SetGlobal("__SOME_KEY__", lua.LString("YYYYYY"))
	assert.Equal(t, lua.LString("YYYYYY"), vm.GetGlobal("__SOME_KEY__")) // Passed
	assert.NoError(t, vm.DoString("assert(__SOME_KEY__ == nil)"))        // Passed, SetGlobal is not working
}
```

When a lua vm has been recycled(in `vmPool.Put(vm)`), `*LState` shoule be considered as reseted as what I got this vm at the first time, but `GetGlobal` is still returning the previous value. Moreover, if I tried to use `SetGlobal` again, the value would not be set in lua like previous.

I have no idea about what was happening......

**Comment by NKUCodingCat** [comment]
it seems that `*LState.Env` is replaced by a new table

https://github.com/ZenLiuCN/glu/blob/5975df7c31b3000f67b653b846630b1509d555d1/pool.go#L85

however, in original implement of `NewState`
![image](https://user-images.githubusercontent.com/8611032/163933725-45ad2adc-3b9c-45da-9e7d-fbe79fb340aa.png)
https://github.com/yuin/gopher-lua/blob/c841877397d8e2ef0bd755390798e6cb957590a9/_state.go#L587

Therefore, I add a line after pool.go#L85 such that
![image](https://user-images.githubusercontent.com/8611032/163935044-1a02e719-4a98-42fd-9587-1252c6749c8f.png)

and everything works now...

**Comment by ZenLiuCN** [comment]
but that is what `s.Polluted` means. `global` designs for not changed by each user scripts.

**Comment by ZenLiuCN** [comment]
The pool should never keep state of a VM unless it's the initial statement, and also means get only instance from pool should the same. in your usecase, should not put into a pool. Plz check something about ObjectPool usecase. if only more questions ,welcom to open this again.
(ps: but our network not always successful to access github)

**Comment by ZenLiuCN** [comment]
take a look again, that may be miss understanded. will check it latter.

**Comment by ZenLiuCN** [comment]
```go
type Global struct {
	MainThread    *LState
	CurrentThread *LState
	Registry      *LTable // this need protect
	Global        *LTable // this also need protect
}
```
coding is done. but some other change not finished. will push latter.

**Comment by ZenLiuCN** [comment]
fix with #c5b66ff
